[new product] Liquibase (#7064)
captn3m0/endoflife.date: products/liquibase.md

---
title: Liquibase
category: framework

> tags: app library amazon apache google ibm microsoft oracle sap vmware

Reviewer: ```suggestion
tags: java-runtime
```

Why these tags, especially `amazon apache google ibm microsoft oracle sap vmware` ?

And considering Liquibase requires java it could be nice to add the `java-runtime` tag.

Note that we have a quite restrictive policy about the use of tags: https://endoflife.date/contribute#new-adding-a-new-product.

Author: Sry, I missed the alphabetic ordering: will correct (in whichever way) when resolved.

I thought it would be a good idea to add those tags as Liquibase is able to handle databases by named vendors (see [here](https://www.liquibase.com/supported-databases)). If this is more intended as "product created by vendor" then I'd kick the tags of course.

Reviewer: Yes, this is intended as "product created by vendor".

iconSlug: liquibase
permalink: /liquibase
versionCommand: liquibase --version
changelogTemplate: https://docs.liquibase.com/start/release-notes/liquibase-release-notes/liquibase-__LATEST__.html

identifiers:
-   repology: liquibase

auto:
  methods:

>   -   git: https://github.com/liquibase/liquibase.git

Reviewer: ```suggestion
  -   maven: org.liquibase/liquibase-core
```

So that only published releases are listed (for exemple this excludes https://github.com/liquibase/liquibase/releases/tag/v4.29.3 which was tagged but never pushed on Maven Central).

# eol(x) = releaseDate(x+1)

> releases:

Reviewer: [4.32 has been released](https://github.com/liquibase/liquibase/releases/tag/v4.32.0) not long ago, can you add it ?

-   releaseCycle: "4.7"
    releaseDate: 2022-01-10
    eol: 2022-02-22
    latest: "4.7.1"
    latestReleaseDate: 2022-01-21

-   releaseCycle: "4.6"
    releaseDate: 2021-11-04
    eol: 2022-01-10
    latest: "4.6.2"
    latestReleaseDate: 2021-12-01

-   releaseCycle: "4.5"
    releaseDate: 2021-09-27
    eol: 2021-11-04
    latest: "4.5.0"
    latestReleaseDate: 2021-09-27

-   releaseCycle: "4.4"
    releaseDate: 2021-06-10
    eol: 2021-09-27
    latest: "4.4.3"
    latestReleaseDate: 2021-08-06

-   releaseCycle: "4.3"
    releaseDate: 2021-02-09
    eol: 2021-06-10
    latest: "4.3.5"
    latestReleaseDate: 2021-04-30

-   releaseCycle: "4.2"
    releaseDate: 2020-11-13
    eol: 2021-02-09
    latest: "4.2.2"
    latestReleaseDate: 2020-12-09

-   releaseCycle: "4.1"
    releaseDate: 2020-09-28
    eol: 2020-11-13
    latest: "4.1.1"
    latestReleaseDate: 2020-10-12

-   releaseCycle: "4.0"
    releaseDate: 2020-07-14
    eol: 2020-09-28
    latest: "4.0.0"
    latestReleaseDate: 2020-07-14

---

> [Liquibase](https://www.liquibase.org/) is an open-source database-independent library for
> tracking, managing and applying database schema changes.

> Liquibase is published under the Apache License 2.0; Liquibase Pro is a commercial extended version

Reviewer: ```suggestion
Liquibase OSS is published under the Apache License 2.0; Liquibase Pro is a commercial extended version
```

To use the same wording as in release notes.

of Liquibase. Both follow the same release cycle running a standard semantic versioning scheme with

> the current release cycle being the only one actively maintained. Breaking changes happen in minor

Reviewer: Is this documented somewhere on https://liquibase.com/, or based on the releases' history ? Can you document it to make it clear ?

Author: I didn't find any explicit documentation about that.

Conclusion was taken from the changelogs, which always list changes for OSS and PRO ([see](https://docs.liquibase.com/start/release-notes/liquibase-release-notes/liquibase-4.31.1.html), or [ancient release years ago](https://docs.liquibase.com/start/release-notes/liquibase-release-notes/liquibase-4.4.2.html), or even the [latest](https://docs.liquibase.com/start/release-notes/liquibase-release-notes/liquibase-4.32.0.html))

Reviewer: Ok, then could you mention that they don't have a support policy and that "the current release cycle being the only one actively maintained" is based on the release history ?

-   releaseCycle: "4.7"
    releaseDate: 2022-01-10
    eol: 2022-02-22
    latest: "4.7.1"
    latestReleaseDate: 2022-01-21

-   releaseCycle: "4.6"
    releaseDate: 2021-11-04
    eol: 2022-01-10
    latest: "4.6.2"
    latestReleaseDate: 2021-12-01

-   releaseCycle: "4.5"
    releaseDate: 2021-09-27
    eol: 2021-11-04
    latest: "4.5.0"
    latestReleaseDate: 2021-09-27

-   releaseCycle: "4.4"
    releaseDate: 2021-06-10
    eol: 2021-09-27
    latest: "4.4.3"
    latestReleaseDate: 2021-08-06

-   releaseCycle: "4.3"
    releaseDate: 2021-02-09
    eol: 2021-06-10
    latest: "4.3.5"
    latestReleaseDate: 2021-04-30

-   releaseCycle: "4.2"
    releaseDate: 2020-11-13
    eol: 2021-02-09
    latest: "4.2.2"
    latestReleaseDate: 2020-12-09

-   releaseCycle: "4.1"
    releaseDate: 2020-09-28
    eol: 2020-11-13
    latest: "4.1.1"
    latestReleaseDate: 2020-10-12

-   releaseCycle: "4.0"
    releaseDate: 2020-07-14
    eol: 2020-09-28
    latest: "4.0.0"
    latestReleaseDate: 2020-07-14

---

> [Liquibase](https://www.liquibase.org/) is an open-source database-independent library for
> tracking, managing and applying database schema changes.

> Liquibase OSS is published under the Apache License 2.0; Liquibase Pro is a commercial extended
> version of Liquibase. Both follow the same release cycle running a standard semantic versioning
> scheme with the current release cycle being the only one actively maintained. Breaking changes
> happen in minor releases too.

Reviewer: ```suggestion
Liquibase OSS is published under the Apache License 2.0;
Liquibase Pro is a commercial extended version of Liquibase.
Both follow the same release cycle running a loosely semantic versioning scheme (Breaking changes may happen in minor releases).
The support policy is not documented, but based on the releases' history, only the latest release is actively maintained.
```

Related to [this discussion](https://github.com/endoflife-date/endoflife.date/pull/7064#discussion_r2117429392), WDYT of this ?

Author: LGTM: will change tonight :+1:

---
title: Liquibase
category: framework

> tags: app java-runtime library

Reviewer: ```suggestion
tags: java-runtime
```

`library` is not an actual endoflife.date tag, and the `framework` category will be also transformed to a tag and is pretty close anyway.

`app` is also a category : this is not allowed [as per our documentation](https://endoflife.date/contribute#new-adding-a-new-product).

Author: hmmm, I don't know how that library sneaked in there :thinking:

fixed :hammer_and_wrench:
